azkaban-common/src/main/java/azkaban/project/DependencyFetcher.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.project;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Properties;
+import javax.xml.bind.annotation.adapters.HexBinaryAdapter;
+import org.apache.log4j.Logger;
+
+import static com.google.common.base.Preconditions.*;
```

Comment:
I agree wildcarding is not good. But it is not always bad.

In this case, this is kinda done intelligently by IntelliJ. If you see these classes, they are designed for static import. Example: Preconditions, Objects, Assert (in Junit).

It is a standard practice to wildcard import classes of these kind so that you can use the utilities easily and not have to write `MyAwesomeLibraryClassName.myUtilMethod()` every time. This in turn avoids multiple static import statements which kinda pollute your import paths.